client/blocks/taxonomy-manager/index.jsx

```diff
@@ -11,27 +12,65 @@ import SectionHeader from 'components/section-header';
 import Card from 'components/card';
 import SearchCard from 'components/search-card';
 import Button from 'components/button';
+import TermsList from './list';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getPostTypeTaxonomy } from 'state/post-types/taxonomies/selectors';
+import QueryTaxonomies from 'components/data/query-taxonomies';
 
 export class TaxonomyManager extends Component {
 	static propTypes = {
-		translate: PropTypes.func,
 		taxonomy: PropTypes.string,
+		labels: PropTypes.object,
+		postType: PropTypes.string,
+		siteId: PropTypes.number,
 	};
 
+	static defaultProps = {
+		postType: 'post'
+	};
+
+	state = {
+		search: null
+	};
+
+	onSearch = searchTerm => {
+		if ( searchTerm !== this.state.search ) {
+			this.setState( {
+				search: searchTerm
+			} );
+		}
+	}
+
 	render() {
+		const { search } = this.state;
+		const { siteId, postType, labels, taxonomy } = this.props;
+		const query = {};
+		if ( search && search.length ) {
+			query.search = search;
+		}
+
 		return (
 			<div>
-				<SectionHeader label={ this.props.translate( 'Categories' ) }>
+				{ siteId && <QueryTaxonomies { ...{ siteId, postType } } /> }
+				<SectionHeader label={ labels.name }>
 					<Button compact primary>
-						{ this.props.translate( 'Add Category' ) }
+						{ labels.add_new_item }
 					</Button>
 				</SectionHeader>
 				<Card>
-					<SearchCard onSearch={ () => {} } />
+					<SearchCard onSearch={ this.onSearch } className="taxonomy-manager__search" />
```

Comment:
We may want to consider debouncing the search, since typing fast is noticeably laggy for me. Fine for a future PR if you'd prefer.